pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -542,6 +546,27 @@ def _set_horovod_backend(self):
         else:
             self.num_processes = hvd.local_size()
 
+    def check_ipython_compatibility(self):
+        """Raises a `MisconfigurationException` if the accelerator is not compatible with IPython and code is run in an IPython kernel."""
+        if self._distrib_type not in self.ipython_compatible_distrib_types:
+            # check ipython env
+            import sys
+            in_ipython = False
+            in_ipython_kernel = False
+            if 'IPython' in sys.modules:
+                from IPython import get_ipython
+                ip = get_ipython()
+                in_ipython = ip is not None
+
+            if in_ipython:
+                in_ipython_kernel = getattr(ip, 'kernel', None) is not None
+            
+            if in_ipython_kernel:
+                raise MisconfigurationException(
+            "Selected distributed backend {self._distrib_type} not compatible with IPython environment"
+            "Run your code as a script, or choose one of compatible backends {self.ipython_compatible_distrib_types} as accelerator backend"
+            )
```

Comment:
needs to be f string otherwise this won't work
also could you run `yapf -i` on this file to get proper formatting?

Reply:
Done and sorry for the mistake